templates/al2023/template.json

```diff
@@ -176,6 +177,7 @@
       "script": "{{template_dir}}/provisioners/install-worker.sh",
       "environment_vars": [
         "AWS_ACCESS_KEY_ID={{user `aws_access_key_id`}}",
+	"AWS_ENDPOINT_URL_S3={{user `aws_endpoint_url_s3`}}",
```

Comment:
The `BINARY_BUCKET_REGION` should be `us-west-2` in govcloud for the time being. I think we need to stop using that variable as a proxy for `AWS_REGION` (the runtime region) when making other partition-specific decisions

Reply:
yeah, now that the region is explicitly set there is no real reason to need to override the S3 endpoint. i've added a commit to remove all the AWS_ENDPOINT_URL_S3 pieces